library/std/src/sys/pal/unix/os.rs

```diff
@@ -193,33 +187,14 @@ pub fn chdir(p: &path::Path) -> io::Result<()> {
     if result == 0 { Ok(()) } else { Err(io::Error::last_os_error()) }
 }
 
-pub struct SplitPaths<'a> {
-    iter: iter::Map<slice::Split<'a, u8, fn(&u8) -> bool>, fn(&'a [u8]) -> PathBuf>,
-}
+pub type SplitPaths<'a> = impl Iterator<Item = PathBuf> + 'a;
```

Comment:
Should these items be `pub(crate)` if so?

We'd rather not be exposing a TAIT to downstream code, I imagine, so it might be better to enforce that with proper visibility:

```suggestion
pub(crate) type SplitPaths<'a> = impl Iterator<Item = PathBuf> + 'a;
```